proto/table_marshal.go

```diff
@@ -2965,6 +2978,17 @@ func (p *Buffer) Marshal(pb Message) error {
 		if _, ok := pb.(Marshaler); ok {
 			return fmt.Errorf("proto: deterministic not supported by the Marshal method of %T", pb)
 		}
+		if _, ok := pb.(reverseMarshaler); ok {
+			return fmt.Errorf("proto: deterministic not supported by the Marshal method of %T", pb)
+		}
+
+	}
+	if m, ok := pb.(reverseMarshaler); ok {
```

Comment:
Currently I think I would prefer too rather not adding another interface here and keep calling the ```XXX``` methods here.
With the direct calling we will skip the ```p.deterministic``` setting on the buffer.

Reply:
I do have a safer approach here: 
https://github.com/gogo/protobuf/pull/627
Keeping the current interfaces/flow.